app/instrumentation/instrumentation.go

```diff
@@ -0,0 +1,59 @@
+package instrumentation
+
+import (
+	"fmt"
+	"log/slog"
+	"net/http"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type instrumentedResponseWriter struct {
+	http.ResponseWriter
+	status int
+}
+
+func (r *instrumentedResponseWriter) WriteHeader(code int) {
+	r.status = code
+	r.ResponseWriter.WriteHeader(code)
+}
+
+// Middleware logs the status code of each request.
+func Middleware(logger *slog.Logger) func(h http.Handler) http.Handler {
+	return func(h http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			iw := instrumentedResponseWriter{w, http.StatusOK}
+
+			h.ServeHTTP(&iw, r)
+
+			route := chi.RouteContext(r.Context()).RoutePattern()
+
+			logger.Info(
```

Comment:
I just noticed this handler isn't really instrumentation, it's logging. You could use something like https://pkg.go.dev/github.com/prometheus/client_golang/prometheus/promhttp#InstrumentHandlerCounter.